src/main/java/org/apache/sling/models/impl/injectors/OSGiServiceInjector.java

```diff
@@ -206,7 +206,7 @@ public void onDisposed() {
                 for (ServiceReference<?> ref : refs) {
                     try {
                         context.ungetService(ref);
-                    } catch (IllegalStateException | IllegalArgumentException | NullPointerException exception) {
+                    } catch (IllegalStateException exception) {
                         log.error("Failed to unget service ref {}", ref, exception);
```

Comment:
I woudln't log an error in this case, as this is an expected case

Reply:
@cziegeler Is there really no need to unget the service references prior to the bundle update? What happens with those dangling service references? Are they automatically released during the bundle update?

Reply:
I haven't looked at what this code is doing, just commenting on catching such exceptions and logging
OSGi cleans up service and service references on bundle events.